Formula/ettercap.rb

```diff
@@ -22,11 +22,22 @@ class Ettercap < Formula
   depends_on "openssl@1.1"
   depends_on "pcre"
 
+  uses_from_macos "curl"
+  uses_from_macos "libpcap"
+  uses_from_macos "ncurses"
+  uses_from_macos "zlib"
+
   def install
     # Work around a CMake bug affecting harfbuzz headers and pango
     # https://gitlab.kitware.com/cmake/cmake/issues/19531
     ENV.append_to_cflags "-I#{Formula["harfbuzz"].opt_include}/harfbuzz"
 
+    # Fix build error on wdg_file.c: fatal error: menu.h: No such file or directory
+    on_linux { ENV.append_to_cflags "-I#{Formula["ncurses"].opt_include}/ncursesw" }
+
+    # Fix missing: GTK3_GLIBCONFIG_INCLUDE_DIR
+    inreplace "cmake/Modules/FindGTK3.cmake", "/usr/local/lib", HOMEBREW_PREFIX/"lib"
```

Comment:
Can we pass this as a `-D` flag to CMake instead of an `inreplace`?

Reply:
Wasn't familiar enough with the CMake logic to see if there was a way.

The inreplace is modifying the function find behavior at
https://github.com/Ettercap/ettercap/blob/v0.8.3.1/cmake/Modules/FindGTK3.cmake#L160-L187

On Linux, the first issue seen was running `_gtk3_find_include_dir(GTK3_GLIBCONFIG_INCLUDE_DIR glibconfig.h)`.

Reply:
Maybe using a default CMake path could help. Just need to check if any work.
I might give `CMAKE_FIND_ROOT_PATH` a try

Reply:
We could probably send a patch to the upstream project adding `/opt/homebrew/lib` to their search paths.

Or maybe even better (for users building from source in a non-default prefix), some CMake code that queries `brew` for `brew --prefix`. Not sure they'll be keen on carrying this code around, in which case we'll want to keep the `inreplace`, but it's worth a shot.

Reply:
I think there are some CMake variables we can use. Still going through documentation in https://cmake.org/cmake/help/latest/command/find_path.html but maybe one of:
- `GTK3_ROOT=#{Formula["gtk+3"]}.opt_prefix`
- `CMAKE_PREFIX_PATH=#{HOMEBREW_PREFIX}`
- `CMAKE_INCLUDE_PATH=#{HOMEBREW_PREFIX}/include`

Reply:
`brew` already sets `CMAKE_PREFIX_PATH` in superenv, so it's probably something else.

Reply:
Passing in the path for the problematic variable does work and will probably have the lowest impact:
```
-DGTK3_GLIBCONFIG_INCLUDE_DIR=#{Formula["glib"].opt_lib}/glib-2.0/include
```

Reply:
Sounds fine to me.